tests/unit/validation/test_input_validation.py

```diff
@@ -550,6 +563,27 @@ def test_validate_three_winding_transformer_ukpkminmax(input_data):
     assert NotGreaterOrEqualError("three_winding_transformer", "pk_23_max", [1], 0) in validation_errors
 
 
+def test_validate_input_data_transformer_tap_regulator(input_data):
+    validation_errors = validate_input_data(input_data, calculation_type=CalculationType.power_flow)
+    assert NotBooleanError("transformer_tap_regulator", "status", [52, 1, 53]) in validation_errors
+    assert (
+        InvalidIdError(
+            "transformer_tap_regulator", "regulated_object", [1], ["transformer", "three_winding_transformer"]
+        )
+        in validation_errors
+    )
+    assert (
+        InvalidEnumValueError("transformer_tap_regulator", "control_side", [1], [BranchSide, Branch3Side])
+        in validation_errors
+    )
+    # TODO (nbharambe) Add control side error after it is included
+    # assert InvalidControlSideError("transformer_tap_regulator", "control_side", [52], BranchSide) in validation_errors
+    assert NotGreaterOrEqualError("transformer_tap_regulator", "u_set", [52], 0) in validation_errors
+    assert NotGreaterThanError("transformer_tap_regulator", "u_band", [52, 1], 0) in validation_errors
+    assert NotGreaterOrEqualError("transformer_tap_regulator", "line_drop_compensation_r", [52], 0) in validation_errors
+    assert NotGreaterOrEqualError("transformer_tap_regulator", "line_drop_compensation_x", [1], 0) in validation_errors
```

Comment:
Just out of random conversation from parallel PR's (https://github.com/PowerGridModel/power-grid-model/pull/620):
```suggestion
    assert NotGreaterOrEqualError("transformer_tap_regulator", "u_set", [52], 0.0) in validation_errors
    assert NotGreaterThanError("transformer_tap_regulator", "u_band", [52, 1], 0.0) in validation_errors
    assert NotGreaterOrEqualError("transformer_tap_regulator", "line_drop_compensation_r", [52], 0.0) in validation_errors
    assert NotGreaterOrEqualError("transformer_tap_regulator", "line_drop_compensation_x", [1], 0.0) in validation_errors
```